src/sentry/event_manager.py

```diff
@@ -1493,8 +1494,10 @@ def _save_aggregate(
                     try:
                         # If the `projects:similarity-embeddings-grouping` feature is disabled,
                         # we'll still get back result metadata, but `seer_matched_group` will be None
-                        seer_response_data, seer_matched_group = get_seer_similar_issues(
-                            event, primary_hashes
+                        seer_response_data, seer_matched_group = with_circuit_breaker(
+                            "event_manager.get_seer_similar_issues",
+                            lambda: get_seer_similar_issues(event, primary_hashes),
+                            options.get("seer.similarity.circuit-breaker-config"),
```

Comment:
where does this handle CircuitBreakerTripped?

Reply:
Yup, just realized this as I was going to post it in the channel. In the middle of fixing it now.

Reply:
Okay, fixed. And actually, we don't want to do anything in that case, because the circuit breaker is really just another check à la those in `should_call_seer_for_grouping`. But it's true that we don't want to capture it as a legit error, so you're right that it needed to be a separate `except` branch from the generic one.

Reply:
BTW, I'm a bit late with that, because you've already merged the CircuitBreaker PR, but I think `with_circuit_breaker` might work better as a context manager, checking if the breaker is open, and returning `CircuitBreakerTripped` if it is, otherwise yielding from within try/except block. Let me check if that's something that I can try quickly.

Reply:
Before you do that, please read over the description in https://github.com/getsentry/sentry/pull/71627. I did consider making it a context manager, but decided people were more likely to use it right if I did it as a function.

Not vetoing switching it, but I'd like to address that concern before we do.